cli/core/findStalePods.go

```diff
@@ -0,0 +1,256 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/core/onchain"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/pkg/errors"
+)
+
+func PodManagerContracts() map[uint64]string {
+	return map[uint64]string{
+		0:     "0x91E677b07F7AF907ec9a428aafA9fc14a0d3A338",
+		17000: "0x30770d7E3e71112d7A6b7259542D1f680a70e315", //testnet holesky
+	}
+}
+
+type Cache struct {
+	PodOwnerShares map[string]PodOwnerShare
+}
+
+type PodOwnerShare struct {
+	Shares     uint64
+	IsEigenpod bool
+}
+
+const ACCEPTABLE_BALANCE_DEVIATION = float64(0.95)
+
+var cache Cache
+
+func isEigenpod(eth *ethclient.Client, chainId uint64, eigenpodAddress string) (bool, error) {
+	if cache.PodOwnerShares == nil {
+		cache.PodOwnerShares = make(map[string]PodOwnerShare)
+	}
+
+	if val, ok := cache.PodOwnerShares[eigenpodAddress]; ok {
+		return val.IsEigenpod, nil
+	}
+
+	// default to false
+	cache.PodOwnerShares[eigenpodAddress] = PodOwnerShare{
+		Shares:     0,
+		IsEigenpod: false,
+	}
+
+	podManAddress, ok := PodManagerContracts()[chainId]
+	if !ok {
+		return false, fmt.Errorf("chain %d not supported", chainId)
+	}
+	podMan, err := onchain.NewEigenPodManager(common.HexToAddress(podManAddress), eth)
+	if err != nil {
+		return false, err
+	}
+
+	if podMan == nil {
+		return false, errors.New("failed to find eigenpod manager")
+	}
+
+	pod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), eth)
+	if err != nil {
+		return false, err
+	}
+
+	owner, err := pod.PodOwner(nil)
```

Comment:
difference between nil and callopts?

Reply:
don't really need callopts here since it's a read call